benchmarks/sharded-bm/bench.sh

```diff
@@ -0,0 +1,285 @@
+    else
+        for node in "${NEAR_HOMES[@]}"; do
+            edit_genesis ${node}/genesis.json
+            edit_config ${node}/config.json
+            edit_log_config ${node}/log_config.json
+        done
+        # set single node RPC port to known value
+        jq --arg val "${RPC_ADDR}" \
+            '.rpc.addr |= $val' ${NEAR_HOMES[-1]}/config.json > tmp.$$.json && mv tmp.$$.json ${NEAR_HOMES[-1]}/config.json || rm tmp.$$.json
+    fi
+    echo "===> Done"
+}
+
+create_accounts() {
+    echo "=> Creating accounts"
+    echo "Number of shards: ${NUM_SHARDS}"
+
+    mkdir -p ${USERS_DATA_DIR}
+
+    num_accounts=$(jq '.num_accounts' ${BM_PARAMS})
+
+    for i in $(seq 0 $((NUM_SHARDS-1))); do
+        prefix=$(printf "a%02d" ${i})
+        data_dir="${USERS_DATA_DIR}/shard${i}"
+        nonce=$((1+i*num_accounts))
+        echo "Creating ${num_accounts} accounts for shard: ${i}, account prefix: ${prefix}, use data dir: ${data_dir}, nonce: ${nonce}"
+        RUST_LOG=info \
+        cargo run --manifest-path ${SYNTH_BM_PATH} --release -- create-sub-accounts \
+            --rpc-url ${RPC_URL} \
+            --signer-key-path ${VALIDATOR_KEY} \
+            --nonce ${nonce} \
+            --sub-account-prefix ${prefix} \
+            --num-sub-accounts ${num_accounts} \
+            --deposit 953060601875000000010000 \
+            --channel-buffer-size 1200 \
+            --interval-duration-micros 800 \
+            --user-data-dir ${data_dir}
+    done
+
+    echo "=> Done"
+}
+
+native_transfers() {
+    echo "=> Running native token transfer benchmark"
+    echo "Number of shards: ${NUM_SHARDS}"
+
+    mkdir -p ${LOG_DIR}
+
+    num_transfers=$(jq '.num_transfers' ${BM_PARAMS})
+    buffer_size=$(jq '.channel_buffer_size' ${BM_PARAMS})
+    interval=$(jq '.interval_duration_micros' ${BM_PARAMS})
+
+    echo "Config: num_transfers: ${num_transfers}, buffer_size: ${buffer_size}, interval: ${interval}"
+
+    trap 'kill $(jobs -p) 2>/dev/null' EXIT
+    for i in $(seq 0 $((NUM_SHARDS-1))); do
+        log="${LOG_DIR}/gen_shard${i}"
+        data_dir="${USERS_DATA_DIR}/shard${i}"
+        echo "Running benchmark for shard: ${i}, log file: ${log}, data dir: ${data_dir}"
+        RUST_LOG=info \
+        cargo run --manifest-path ${SYNTH_BM_PATH} --release -- benchmark-native-transfers \
+            --rpc-url ${RPC_URL} \
+            --user-data-dir ${data_dir}/ \
+            --num-transfers ${num_transfers} \
+            --channel-buffer-size ${buffer_size} \
+            --interval-duration-micros ${interval} \
+            --amount 1 &> ${log} &
+    done
+    wait
+    
+    echo "=> Done"
+}
+
+monitor() {
+    old_now=0
+    old_processed=0
+    while true 
+    do 
+        date
+        now=$(date +%s%3N)
+        processed=$(curl -s localhost:3030/metrics | grep transaction_processed_total | grep -v "#" | awk '{ print $2 }')
```

Comment:
You might want `near_transaction_processed_successfully_total` or [near_action_called_count](https://github.com/near/nearcore/pull/12874#discussion_r1956348619).

Reply:
Good point, thanks!